wallet/src/de/schildbach/wallet/livedata/EncryptWalletLiveData.kt

```
@@ -43,18 +43,33 @@ class EncryptWalletLiveData(
     fun savePin(pin: String) {
-        securityGuard.removeKeys()
-        securityGuard.savePin(pin)
-        securityGuard.ensurePinFallback(pin)
         val wallet = walletApplication.wallet!!
         if (wallet.isEncrypted) {
+            // a repeated invocation can race the encrypt task (UI re-entry); if the
+            // wallet was already encrypted using this same PIN, the first invocation
+            // did all the work and this one is a harmless no-op
+            val alreadySavedSamePin = try {
+                securityGuard.checkPin(pin)
+            } catch (x: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)